packages/amplify-graphql-transformer-core/src/transformer-context/resource-helper.ts

```diff
@@ -0,0 +1,31 @@
+import { GraphQLApiProvider, TransformerResourceHelperProvider } from '@aws-amplify/graphql-transformer-interfaces';
+import { CfnParameter } from '@aws-cdk/core';
+import { StackManager } from './stack-manager';
+
+export class TransformerResourceHelper implements TransformerResourceHelperProvider {
+  private api?: GraphQLApiProvider;
+  // eslint-disable-next-line no-useless-constructor
+  constructor(private stackManager: StackManager) {}
+  public generateResourceName = (name: string): string => {
+    if (!this.api) {
+      throw new Error('API not initalized');
+    }
+    this.ensureEnv();
+    const env = (this.stackManager.getParameter('env') as CfnParameter).valueAsString;
+    const apiId = this.api!.apiId;
+    return `${name}-${apiId}-${env}`;
+  };
+
+  bind(api: GraphQLApiProvider) {
+    this.api = api;
+  }
+
+  private ensureEnv = (): void => {
```

Comment:
Can we have some logic to only do the legacy `NONE` env support for legacy scenarios and finally get rid of it?

Reply:
Done!